src/org/sipdroid/sipua/SipdroidEngine.java

```diff
@@ -453,23 +483,28 @@ public void register() {
 					i++;
 					continue;
 				}
-				user_profiles[i].contact_url = getContactURL(user_profiles[i].from_url,sip_providers[i]);
-		
+				user_profiles[i].contact_url = getContactURL(user_profiles[i].from_url, sip_providers[i]);
+
 				if (!Receiver.isFast(i)) {
 					unregister(i);
 				} else {
 					if (ra != null && ra.register()) {
-						Receiver.onText(Receiver.REGISTER_NOTIFICATION+i,getUIContext().getString(R.string.reg),R.drawable.sym_presence_idle,0);
+						Receiver.onText(Receiver.REGISTER_NOTIFICATION + i, getUIContext().getString(R.string.reg), R.drawable.sym_presence_idle, 0);
 						wl[i].acquire();
 					}
 				}
 			} catch (Exception ex) {
-				
+
 			}
 			i++;
 		}
+		for (PublishAgent pa : pas) {

```

Comment:
@dpocock What would be the advantage of doing that? In that case I need to find all the parent function from where 'ras' is being implemented and call publish()
Should I do that?